ssl/ssl_rsa.c

```diff
@@ -1009,8 +1009,9 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
         extension = NULL;
     }
 
-    ret = SSL_CTX_use_serverinfo_ex(ctx, version, serverinfo,
-                                    serverinfo_length);
+    if (version != 0)
```

Comment:
Yeah. Big difference. OPENSSL_assert() kills the process even in production builds. We've had CVEs from reachable asserts in production code, e.g. if an attacker can cause some malformed data to cause the assert to be reached, then they can kill the process. Most likely an error of this sort is not fatal for the whole process.